lib/tapioca/generator.rb

```diff
@@ -522,5 +543,32 @@ def compile_dsl_rbi(constant, contents)
 
       filename
     end
+
+    sig { params(tmp_dir: Pathname).returns(T.nilable(String)) }
```

Comment:
Could be an array of errors for you can return all of them, not just the first one.

Reply:
I actually build the error message within the method. So we return `New file(s) added.` or a list of modified files.

Reply:
I think this method does too much, it shouldn't be controlling the message.

What do you think about returning a struct like this instead?

```rb
class DslState < T::Struct
  new_files, T::Array[String]
  update_files, T::Array[String]

  def dirty?
    !new_files.empty? || !updated_files.empty?
  end
end
```

So the `verify_dsl_rbi` method builds the current state and the client decide what to do with the information?

Reply:
I'm not opposed to that at all, but I'm going to opt to leaving that to the future refactors @paracycle and I discussed.

Reply:
Agreed that this can be better, but I think this whole file needs a refactor anyway, so I am fine with the current state.